### DISCUSSION TITLE
Can custom annotations be made?

### DISCUSSION THREAD

**Discussion by mor-anhln** [chartjs/chartjs-plugin-annotation#936] [Q&A] [answered]
I want to custom an annotation like above, this image must be generated based on data so I can't use an image, is there any way to do it?
![image](https://github.com/user-attachments/assets/8b9fba63-5c7a-4e09-97ba-952e89ebd6e3)

**Comment by mor-anhln** [comment]
@stockiNail help me

**Comment by stockiNail** [comment, marked as answer]
@anhln1998 there is not any support for custom annotation. This was evaluated a couple of years ago but was discarded.

To do what you are showing, I would implement a label annotation setting a canvas as content (https://www.chartjs.org/chartjs-plugin-annotation/latest/samples/label/canvas.html).
The label annotation could have text, image or canvas content. 
You could create a a canvas (pay attention to the dimension), you can draw the shape you want and you can set it to the label annotation (content option).
There are other specific option to act on canvas (like width, height).

Otherwise, if too complex, I would implement a specific ChartJS plugin to do that.
Hopefully this could help you.

**Reply by mor-anhln** [reply]
@stockiNail thank you
